src/qat/ir/instructions.py

```diff
@@ -0,0 +1,1032 @@
+        axes=[],
+        args=[],
+        results_needed: bool = False,
+        **kwargs,
+    ):
+        # TODO: Should find out why ags can't reference model objects in builder serialization.
+        #   But deep copying them, as the values shouldn't vary, should be fine.
+
+        super().__init__(
+            acquire=acquire,
+            process=process,
+            args=args or [],
+            axes=[] if axes == None else axes if isinstance(axes, List) else [axes],
+            results_needed=results_needed,
+        )
+
+    @property
+    def output_variable(self):
+        return self.acquire.output_variable
+
+    def __repr__(self):
+        axis = ",".join([axi.value for axi in self.axes])
+        args = f",{','.join(str(arg) for arg in self.args)}" if len(self.args) > 0 else ","
+        output_var = f"->{self.output_variable}" if self.output_variable is not None else ""
+        return (
+            f"{self.process.value} {self.acquire.output_variable}{args}{axis}{output_var}"
+        )
+
+
+class Reset(QuantumInstruction):
+    """Resets this qubit to its starting state."""
+
+    inst: Literal["Reset"] = "Reset"
+
+    def __init__(self, quantum_targets, **kwargs):
+        return super().__init__(quantum_targets=quantum_targets)
+
+    def __repr__(self):
+        return f"reset {','.join(self.quantum_targets)}"
+
+    @field_validator("quantum_targets", mode="before")
+    @classmethod
+    def _components_to_ids(cls, targets):
+        if not isinstance(targets, (set, list, tuple)):
+            targets = [targets]
+
+        invalid_reset_targets = [
+            str(val) for val in targets if not isinstance(val, (str, Qubit, PulseChannel))
+        ]
+        if any(invalid_reset_targets):
+            raise ValueError(
+                "Tried to reset on non-qubit/pulse channel "
+                f"{', '.join(invalid_reset_targets)}."
+            )
+        targets = [
+            (
+                val.get_drive_channel().full_id()
+                if isinstance(val, Qubit)
+                else (val.full_id() if isinstance(val, PulseChannel) else val)
+            )
+            for val in targets
+        ]
+        return targets
+
+
+class DeviceUpdate(QuantumInstruction):
+    """
+    Dynamically assigns a value to a particular symbol or hardware attribute during
+    execution.
+
+    .. note:: It's still unknown how this will be represented in the instructions themselves, but that'll come later.
+    For now we perform programatic modification and a before/after state.
+    """
+
+    inst: Literal["DeviceUpdate"] = "DeviceUpdate"
+    attribute: str
+    value: Any
+
+    def __init__(self, target: QuantumComponent, attribute: str, value, **kwargs):
+        super().__init__(target, attribute=attribute, value=value)
+
+    def __repr__(self):
+        return f"{self.target}.{self.attribute} = {str(self.value)}"
+
+
+### Instruction blocks
+def _all_instructions(start_types=[Instruction]):
```

Comment:
Redefining of a function in `instruction_list.py`. Even if it's a fairly short function there's risk of inconsistencies creeping in.